extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultRecorder.java

```diff
@@ -18,12 +18,10 @@
 
     private static final Logger log = Logger.getLogger(VaultRecorder.class);
 
-    public RuntimeValue<ConfigSourceProvider> configureRuntimeProperties(VaultBuildTimeConfig vaultBuildTimeConfig,
-            VaultRuntimeConfig vaultRuntimeConfig,
-            TlsConfig tlsConfig) {
-
+    public RuntimeValue<ConfigSourceProvider> configureRuntimeProperties(VaultRuntimeConfig vaultRuntimeConfig) {
         if (vaultRuntimeConfig.url.isPresent()) {
-            VaultManager.init(vaultBuildTimeConfig, vaultRuntimeConfig, tlsConfig);
+            Arc.container().instance(VaultAuthManager.class).get().setVaultRuntimeConfig(vaultRuntimeConfig);
```

Comment:
@vsevel I think the best way to access Arc is via `io.quarkus.arc.deployment.BeanContainerBuildItem` - you'd add it as the built step parameter and it will give you `BeanContainer` which will go to the recorder, try it please

Reply:
thanks for taking a look. happy christmas and new year as well!
I tried the `BeanContainerBuildItem` approach. just passing the `BeanContainer` instance to the `configureRuntimeProperties()` method (not actually using it), messes up the initialization. it may be related to the fact that the `VaultRuntimeConfig` is running  in the bootstrap phase? I do not see any other reason. but it makes one of the test fail because of a missing property that should have been loaded by the vault config source.
however doc says:
```
However, it is safe to expect that the CDI container is fully initialized and running in a `@Record(RUNTIME_INIT)` build step.
You can obtain a reference to the container via `CDI.current()` or Quarkus-specific `Arc.container()`.
```
so I think the current approach is acceptable.
I will look anyway at the `synthetic bean` approach.

Reply:
here is the code that makes the test fail:
```
    public RuntimeValue<ConfigSourceProvider> configureRuntimeProperties(VaultRuntimeConfig vaultRuntimeConfig,
            BeanContainer beanContainer) {
        if (vaultRuntimeConfig.url.isPresent()) {
            Arc.container().instance(VaultAuthManager.class).get().setVaultRuntimeConfig(vaultRuntimeConfig);
            Arc.container().instance(OkHttpVaultClient.class).get().initWith(vaultRuntimeConfig);
            return new RuntimeValue<>(new VaultConfigSourceProvider(vaultRuntimeConfig));
        } else {
            return emptyRuntimeValue();
        }
    }
```

Reply:
@vsevel sure if the current code works then it is fine. Sorry, it may indeed be the issue of the different phases mix-up.